onnxruntime/core/providers/qnn/qnn_execution_provider.cc

```diff
@@ -274,28 +336,61 @@ Status QNNExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>& fused
     LOGS(logger, VERBOSE) << "fused node name: " << fused_node.Name();
     qnn_models_.emplace(fused_node.Name(), std::move(qnn_model));
 
-    NodeComputeInfo compute_info;
-    compute_info.create_state_func = [&](ComputeContext* context, FunctionState* state) {
-      LOGS(logger, VERBOSE) << "compute_info.create_state_func context->node_name: " << context->node_name;
-      *state = qnn_models_[context->node_name].get();
-      return 0;
-    };
+    ORT_RETURN_IF_ERROR(CreateComputeFunc(node_compute_funcs, logger));
+  }
+  return Status::OK();
+}
 
-    compute_info.release_state_func = [](FunctionState state) {
-      // the 'state' is a qnn::QnnModel managed by unique_ptr
-      ORT_UNUSED_PARAMETER(state);
-    };
+bool QNNExecutionProvider::IsContextCacheFileExists(const onnxruntime::PathString& model_pathstring,
+                                                    onnxruntime::PathString& context_cache_pathstring) const {
+  // Use user provided context cache file path if exist, otherwise try model_file.onnx.bin by default
+  if (!context_cache_path_.empty()) {
+    context_cache_pathstring = ToPathString(context_cache_path_);
+  } else {
+    context_cache_pathstring = model_pathstring + ToPathString(".bin");
+  }
+  std::fstream context_cache_fs(context_cache_pathstring.c_str());
+  bool context_cache_file_exist = context_cache_fs.good() ? true : false;
+  context_cache_fs.close();
 
-    compute_info.compute_func = [](FunctionState state, const OrtApi*, OrtKernelContext* context) {
-      Ort::KernelContext ctx(context);
-      qnn::QnnModel* model = reinterpret_cast<qnn::QnnModel*>(state);
-      Status result = model->ExecuteGraph(ctx);
-      return result;
-    };
+  return context_cache_file_exist;
+}
 
-    node_compute_funcs.push_back(compute_info);
+Status QNNExecutionProvider::Compile(const std::vector<FusedNodeAndGraph>& fused_nodes_and_graphs,
+                                     std::vector<NodeComputeInfo>& node_compute_funcs) {
+  const auto& logger = *GetLogger();
+  bool is_npu_backend = qnn_backend_manager_->IsNpuBackend();
+
+  if (context_cache_enabled_) {
+    Node& fused_node = fused_nodes_and_graphs[0].fused_node;
```

Comment:
Do we need to enforce that the entire model is supported in one single partition? That is, should GetCapability() return an error if we're using context caching and QNN EP can't place all nodes into a single partition?